### PR TITLE
Installation: update setup.py and cleanup docker files

### PR DESCRIPTION
**Pull request by mloubout**
Add mpi and nvidia options to setup.py

Small cleanup of docker file to actually install devito rather than setting python path for every command

### CODE REVIEW

File: docker/Dockerfile.nvidia [hunk 1]
```diff
@@ -125,16 +116,14 @@ RUN if [ "x$MPI_VER" = "x4" ]; then \
...
-ADD ./examples /app/examples
-ADD ./benchmarks /app/benchmarks
-COPY setup.cfg /app/
+# install
+RUN python3 -m venv /venv && \
+    /venv/bin/pip install --no-cache-dir --upgrade pip && \
+    /venv/bin/pip install --no-cache-dir wheel && \
+    /venv/bin/pip install --no-cache-dir -e /app/devito[extras,mpi,nvidia]
+
+ENV CFLAGS=
```
Comment (georgebisbas) [inline docker/Dockerfile.nvidia:127]: ?

File: docker/Dockerfile.nvidia [hunk 2]
```diff
@@ -125,16 +116,14 @@ RUN if [ "x$MPI_VER" = "x4" ]; then \
...
-    rm -rf ~/.cache/pip
-ENV CFLAGS=
 
-ADD ./devito /app/devito
-ADD ./tests /app/tests
-ADD ./scripts /app/scripts
-ADD ./examples /app/examples
-ADD ./benchmarks /app/benchmarks
-COPY setup.cfg /app/
+# install
```
Comment (georgebisbas) [inline docker/Dockerfile.nvidia:121]: Maybe enrich...
# Upgrade pip, install wheel, devito and rest....

File: docker/Dockerfile.nvidia [hunk 3]
```diff
@@ -178,7 +167,7 @@ RUN mkdir -p /app && \
     
 # Change to the app user.
 USER app
-WORKDIR /app
+WORKDIR /app/devito
```
Comment (georgebisbas) [inline docker/Dockerfile.nvidia:171]: I guess we are not anymore backwards compatible with any people that will be developing or running scripts upon docker installation?

**Review by georgebisbas — COMMENTED** [review]
Nice cleanup

File: docker/Dockerfile.nvidia (hunk 3 above)
Comment (mloubout) [reply docker/Dockerfile.nvidia:171]: We are the root dir is different but the structure is the same

File: docker/Dockerfile.nvidia (hunk 1 above)
Comment (mloubout) [reply docker/Dockerfile.nvidia:127]: Was already there

File: docker/Dockerfile.nvidia [hunk 4]
```diff
@@ -125,16 +117,14 @@ RUN if [ "x$MPI_VER" = "x4" ]; then \
...
                $HPCSDK_HOME/comm_libs/mpi ; \
     fi;  \
-    /venv/bin/pip install --no-cache-dir -r /app/requirements-mpi.txt && \
-    rm -rf ~/.cache/pip
-ENV CFLAGS=
 
-ADD ./devito /app/devito
-ADD ./tests /app/tests
-ADD ./scripts /app/scripts
-ADD ./examples /app/examples
```
Comment (FabioLuporini) [inline docker/Dockerfile.nvidia:135]: How are the examples/benchmarks now brought in?

**Review by FabioLuporini — COMMENTED** [review]
nice cleanup! I just have a minor question about the examples and benchmarks folders that my sleepy brain is still failing to catch

File: docker/Dockerfile.nvidia [hunk 5]
```diff
@@ -79,18 +79,10 @@ ENV CUDA_ROOT $HPCSDK_HOME/cuda/bin
...
-    /venv/bin/pip install --no-cache-dir --upgrade pip && \
-    /venv/bin/pip install --no-cache-dir wheel && \
-    /venv/bin/pip install --no-cache-dir -r /app/requirements.txt && \
-    /venv/bin/pip install --no-cache-dir -r /app/requirements-optional.txt && \
-    /venv/bin/pip install --no-cache-dir -r /app/requirements-nvidia.txt && \
-    rm -rf ~/.cache/pip
+# Devito
+ADD . /app/devito
+# Remove git files
+RUN rm -rf /app/devito/.git
```
Comment (FabioLuporini) [inline docker/Dockerfile.nvidia:85]: why?

**Comment by FabioLuporini** [comment]
PR name also needs a suitable tag

File: docker/Dockerfile.nvidia (hunk 4 above)
Comment (mloubout) [reply docker/Dockerfile.nvidia:135]: Same as it was and as it is in devito, this was just copying each subdir one by one

File: docker/Dockerfile.nvidia (hunk 5 above)
Comment (mloubout) [reply docker/Dockerfile.nvidia:85]: Git history takes space and the docker user doesn't have a git account anyway

File: docker/Dockerfile.nvidia (hunk 2 above)
Comment (mloubout) [reply docker/Dockerfile.nvidia]: Fixed

**Pull request merged**